design/multiple-storage-types-support.md

```diff
@@ -288,6 +312,14 @@ type Device struct {
 After Kubernetes has done the general deserialization of the `StorageScopeSpec` into a strong type with weakly typed maps of backend specific config properties, the Ceph operator could easily convert this map into a strong config type that is has knowledge of.
 Other backend operators could do a similar thing for their node/device level config.
 
+#### Placement, Resources, Network
+
+As previously mentioned, the `rook.io` API group will also define some other useful `*Spec` types:
+
+* `PlacementSpec`: Defines placement requirements for components of the storage provider, such as node and pod affinity.  This is similar to the existing [Ceph focused `PlacementSpec`](https://github.com/rook/rook/blob/release-0.7/pkg/apis/rook.io/v1alpha1/types.go#L141), but in a generic way that is reusable by all storage providers.  A `PlacementSpec` will essentially be a map of placement information structs that are indexed by component name.
```

Comment:
placement, network, and resource specs are fairly simple. Do we have any more complex examples of the shared types?

Reply:
Even 'PlacementSpec' is tricky. NexentaEdge is planning on supporting Placement-impacting metadata at any scope, from Tenant through Bucket, and folder or even down to a specific object version. We could map this to the Tenant Scope, but it is only setting a *default*, not a firm specification.

Reply:
@travisn the `StorageScopeSpec` is the most complex one so far.  I'd like for it to abstract some logic for providers also such as device names changing across reboots.

@caitlinbestler all of these `*Spec` types will be optional, none of them are required for a storage provider.